src/RadhydroSimulation.hpp

```diff
@@ -1026,6 +1026,18 @@ auto RadhydroSimulation<problem_t>::advanceHydroAtLevel(amrex::MultiFab &state_o
 			if (Verbose()) {
 				amrex::Print() << "[FOFC-1] flux correcting " << ncells_bad << " cells on level " << lev << "\n";
 				const amrex::IntVect cell_idx = redoFlag.maxIndex(0);
+				// Calculate the coordinates based on the cell index and cell size
+				amrex::Real x_coord = geom[lev].ProbLo(0) + (cell_idx[0] + 0.5) * geom[lev].CellSize(0);
+				amrex::Real y_coord = NAN;
+				amrex::Real z_coord = NAN;
+				if (AMREX_SPACEDIM > 1) {
+					y_coord = geom[lev].ProbLo(1) + (cell_idx[1] + 0.5) * geom[lev].CellSize(1);
+					if (AMREX_SPACEDIM > 2) {
+						z_coord = geom[lev].ProbLo(2) + (cell_idx[2] + 0.5) * geom[lev].CellSize(2);
+					}
+				}
+
+				amrex::Print() << x_coord << ", " << y_coord << ", " << z_coord << ": ";
```

Comment:
Can you put this in a separate function? I think that would be better than duplicating the code.